numba/core/typing/templates.py

```diff
@@ -246,7 +246,24 @@ def fold_arguments(pysig, args, kws, normal_handler, default_handler,
     return args
 
 
-class FunctionTemplate(object):
+class _Template(object):
+
+    def get_template_info(self):
```

Comment:
Why can't this be in `FunctionTemplate`? I don't see other class inheriting from `_Template`.

Reply:
It can be I think in hindsight, at first, wasn't so sure as there appeared to be some strange inheritance.

Reply:
3ecedbb